packages/stores/src/account/osmosis/index.ts

```diff
@@ -1605,7 +1606,223 @@ export class OsmosisAccountImpl {
   }
 
   /**
-   * https://docs.osmosis.zone/developing/modules/spec-lockup.html#lock-tokens
+   * Allows a user to unbond and convert GAMM shares in a lock or available GAMM shares to staked OSMO.
+   *
+   * @param convertibleAssets Assets to unbond. Can be either locked shares or unlocked gamm shares.
+   * @param validatorAddress Bech32 address of validator to delegate to, if not delegating to validator set.
+   * @param maxSlippage Max tolerated slippage for swaps to staking token. Default: 2.5.
+   * @param memo Transaction memo.
+   * @param onFulfill Callback to handle tx fullfillment given raw response.
+   */
+  async sendUnbondAndConvertToStakeMsgs(
+    convertibleAssets: (
+      | { lockId: string }
+      | { availableGammShare: CoinPretty }
+    )[],
+    validatorAddress?: string,
+    maxSlippage = DEFAULT_SLIPPAGE,
+    memo = "",
+    onFulfill?: (tx: DeliverTxResponse) => void
+  ) {
+    const queryAccountLocked = this.queries.queryAccountLocked.get(
+      this.address
+    );
+    await queryAccountLocked.waitFreshResponse();
+
+    const maxSlippageDec = new Dec(maxSlippage).quo(
+      DecUtils.getTenExponentNInPrecisionRange(2)
+    );
+
+    const stakeCurrency = this.chainGetter.getChain(this.chainId).stakeCurrency;
+
+    const involvedQueryPools: ObservableQueryPool[] = [];
+
+    /** Locally relevant function for calculating the minimum amount of staking token to stake by converting
+     *  GAMM shares and swapping the GAMM shares in the pool to the staking token.
+     *  Provided to msg for slippage and price impact protection of user. */
+    const calcMinAmtToStake = async (shares: CoinPrimitive): Promise<Int> => {
```

Comment:
This whole function is pretty long. What about extracting calcMinAmtToStake and then just referenced in the sendUnbondAndConvertToStake ?

Reply:
Rationale for placement here:
* Given that it's a closure, it contains some vars from the immediate scope
* I have high confidence that the logic is only relevant and useful within this function, given how bespoke it is

Let me know what you think given the above points.